File: pset6-py/cash/cash.py
def main():
    # Get dollars from user
    while True:
        try:
            dollars = float(input("Change owed: "))
            if dollars >= 0:
                break
        except ValueError:
            pass
    
    # Convert to cents and round to avoid floating point errors
    cents = round(dollars * 100)
    
    # Count coins
    coins = 0
    for coin_value in [25, 10, 5, 1]:  # quarters, dimes, nickels, pennies
        coins += cents // coin_value
        cents %= coin_value
    
    print(coins)

if __name__ == "__main__":
    main()

File: pset1-c/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    float dollars;
    do
    {
        dollars = get_float("Change owed: ");
    }
    while (dollars < 0);

    // Convert to cents and round to avoid floating point errors
    int cents = round(dollars * 100);

    int coins = 0;
    int denoms[] = {25, 10, 5, 1};
    for (int i = 0; i < 4; i++)
    {
        coins += cents / denoms[i];
        cents %= denoms[i];
    }

    printf("%i\n", coins);
}
